airbyte-integrations/connectors/source-surveymonkey/source_surveymonkey/streams.py

```diff
@@ -0,0 +1,265 @@
+        params = super().request_params(stream_state=stream_state, **kwargs)
+        params["sort_order"] = "ASC"
+        params["sort_by"] = "date_modified"
+        params["per_page"] = 1000  # maybe as user input or bigger value
+        since_value = pendulum.parse(stream_state.get(self.cursor_field)) if stream_state.get(self.cursor_field) else self._start_date
+
+        since_value = max(since_value, self._start_date)
+        params["start_modified_at"] = since_value.strftime("%Y-%m-%dT%H:%M:%S")
+        return params
+
+    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> Mapping[str, Any]:
+        """
+        Return the latest state by comparing the cursor value in the latest record with the stream's most recent state object
+        and returning an updated state object.
+        """
+        latest_state = latest_record.get(self.cursor_field)
+        current_state = current_stream_state.get(self.cursor_field) or latest_state
+        if current_state:
+            return {self.cursor_field: max(latest_state, current_state)}
+        return {}
+
+
+class Surveys(IncrementalSurveymonkeyStream):
+    """
+    Docs: https://developer.surveymonkey.com/api/v3/#surveys
+    A source for stream slices. It does not contain useful info itself.
+    """
+
+    cursor_field = "date_modified"
+
+    def path(self, **kwargs) -> str:
+        return "surveys"
+
+    def parse_response(self, response: requests.Response, stream_state: Mapping[str, Any], **kwargs) -> Iterable[Mapping]:
+        response_json = response.json()
+        if response_json.get("error"):
+            raise Exception(response_json.get("error"))  # TODO: apply ShouldRetry
+        result = response_json.get(self.data_field, [])
+        for record in result:
+            substream = SurveyDetails(survey_id=record["id"], start_date=self._start_date, authenticator=self.authenticator)
+            child_record = substream.read_records(sync_mode=SyncMode.full_refresh)
+            yield from child_record
+
+
+class SurveyDetails(SurveymonkeyStream):
+    """
+    Actually stream above can just filter, sort and enumarate survey ids.
+    It does not contain the needed information. I will need it as stream slice source
+    in all of endpoints
+    """
+
+    def __init__(self, survey_id, start_date, **kwargs):
+        self.survey_id = survey_id
+        super().__init__(start_date=start_date, **kwargs)
+
+    def path(self, stream_slice: Mapping[str, Any] = None, **kwargs) -> str:
+        return f"surveys/{self.survey_id}/details"
+
+    def parse_response(self, response: requests.Response, stream_state: Mapping[str, Any], **kwargs) -> Iterable[Mapping]:
+        response_json = response.json()
+        if response_json.get("error"):
+            raise Exception(response_json.get("error"))  # TODO: apply ShouldRetry
+        output = {k: v for k, v in response_json.items() if k != "pages"}
```

Comment:
Let's not used name like "k" and "v".

Reply:
explanation about why I use SurveyDetails inside Surveys.

The surveys stream returns the response containing objects. Each object have fields `title`, `nickname` and `href` - only. It even does not contain the cursor field. Cursor field and other data can be get only directly through `/surveys/<survey_id>/details`. Details also contan `title, nickname and href` . In a user point of view stream surveys is useless, all the data we get actually from details, which contains expanded data. However, I need the surveys stream to enumerate survey_id to use it in all others streams. This implementation related to necessity both use enumeration and expand the data from child stream.